src/test/groovy/unit/org/pih/warehouse/inventory/TransactionTypeSpec.groovy

```diff
@@ -0,0 +1,91 @@
+package org.pih.warehouse.inventory
+
+import spock.lang.Unroll
+
+import grails.testing.gorm.DomainUnitTest
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class TransactionTypeSpec  extends Specification implements DomainUnitTest<TransactionType> {
+
+
+    @Test
+    @Unroll('TransactionType.validate() with name: #value should return #expected with errorCode: #expectedErrorCode')
+    void 'name field validation'() {
+        when:
+        domain.name = value
+
+        then:
+        assert domain.validate(['name']) == expected
+        assert domain.errors.getFieldError("name")?.getCode() == expectedErrorCode
+
+        where:
+        value                                             || expected | expectedErrorCode
+        null                                              || false    | 'nullable'
+        'this_is_big_string_to_test_max_length_255' * 255 || false    | 'maxSize.exceeded'
+        'Dummy Name'                                      || true     | null
+    }
+
+    @Test
+    @Unroll('TransactionType.validate() with description: #value should return #expected with errorCode: #expectedErrorCode')
+    void 'description field validation'() {
+        when:
+        domain.description = value
+
+        then:
+        assert domain.validate(['description']) == expected
+        assert domain.errors.getFieldError("description")?.getCode() == expectedErrorCode
+
+        where:
+        value                                             || expected | expectedErrorCode
+        null                                              || true     | null
+        'this_is_big_string_to_test_max_length_255' * 255 || false    | 'maxSize.exceeded'
+        'Dummy description'                               || true     | null
+    }
+
+    @Test
+    void 'all fields validation'() {
```

Comment:
what does the empty domain do here? What case is this testing?

Reply:
we are testing for empty domain. so it can throw error for all fields that has some validation. So in future if add any new field that can be catch here.

Reply:
Basically, so we can check all of the errors at once in case we accidentally added a nullable:false validation that isn't being tested elsewhere. This should also catch validation that was removed. We should make sure this test includes a check on the actual fields / error codes returned in the Errors object, not just the count.

Reply:
okay that makes sense. I like the idea. It forces you to consider tests when you make a schema change. Though I agree it's probably a good idea to at least assert on each of the errors instead of just the number of errors

Reply:
Yes, that makes sense as you may have removed one validation rule and added another.